clients/venice-producer/src/main/java/com/linkedin/venice/producer/online/ProducerTool.java

```diff
@@ -193,17 +196,23 @@ private static void writeToStore(ProducerContext producerContext) throws Excepti
 
   private static Object getValueObject(String valueString, RouterBasedStoreSchemaFetcher schemaFetcher) {
     Object value = null;
+    List<Exception> exceptionList = new ArrayList<>();
     for (Schema valueSchema: schemaFetcher.getAllValueSchemas()) {
       try {
         value = adaptDataToSchema(valueString, valueSchema);
         break;
       } catch (Exception e) {
-        // Nothing to do. Try the next schema
+        exceptionList.add(e);
```

Comment:
can we add only the top most or top few stacks ? rest of them may not provide any valuable info.

Reply:
This messaging is for the user and the stack trace doesn't provide any valuable information for them. We can add a debug mode if we want more details for debugging.